frontend/ui/data/css_tweaks.lua

```diff
@@ -1026,11 +1044,54 @@ This tweak can be duplicated as a user style tweak when books contain footnotes
 {
     -cr-hint: footnote-inpage;
     margin: 0 !important;
-    font-size: 0.8rem !important;
+}
+*, autoBoxing {
+    -cr-hint: late;
+    -cr-only-if: inpage-footnote;
+        font-size: 0.8rem !important;
+}
+*, autoBoxing {
+    -cr-hint: late;
+    -cr-only-if: extended-inpage-footnote;
+        font-size: 0.8rem !important;
 }
             ]],
             separator = true,
         },
+        {
+            title = _("In-page footnote extension"),
+            {
+                id = "following-footnote-inpage_all",
+                conflicts_with = function(id) return util.stringStartsWith(id, "following-footnote-inpage_") end,
+                title = _("Extend footnote content until next entry"),
+                description = _([[
+Extend in-page footnotes shown at the bottom of pages to include text up to the next footnote.
+This is needed when books don't correctly mark all text that belongs to the footnote.]]),
+                css = [[
+*, autoBoxing {
+    -cr-only-if: following-inpage-footnote -inpage-footnote;
+        -cr-hint: extend-footnote-inpage;
```

Comment:
It feels that this, having a really low specificity, may not match those that get to be inpage-footnotes with `*[type~="note"]` or `.footnote` which has a higher specificity.
Is `-cr-hint: late` the solution ? Dunno how everything behave when multiple cr-hint late setting other hints... moreover when one would use `-cr-only-if: extended-inpage-footnote`...

Reply:
It's certainly more fragile, I'll do some testing with `-cr-hint: late`.

`-cr-only-if: extended-inpage-footnote` is essentially only needed to set the smaller font-size above for well behaved books where you don't need to force a single size via the `-cr-only-if: inside-inpage-footnote` hint.

We could also go back and make `-cr-only-if: inpage-footnote` also match extended footnotes. Less css but also less flexible. I at least found it useful in debugging to match the extended blocks specifically.

Reply:
Logically, `-cr-only-if: inpage-footnote` (named `inpage-foonote`, different from the hint name `footnote-inpage`) should match what has ended up being a footnote, not what had the hint.
So yes, I think it should also match extended footnotes.

Same for `inside-inpage-footnote`, which should match stuff inside an extended footnote block.

(I'm out for the day.)

Reply:
I agree, changed it here and dropped the unneeded `-cr-only-if: extended-inpage-footnote`: https://github.com/koreader/crengine/pull/620

Reply:
I remembered why I didn't originally add `late`: I had problems with specificity like you hinted at.

I did what I think is the correct fix now by using `-cr-hint: late` to ensure it's applied after footnotes are marked as well as `:where()` and `priority` to ensure it's lower specificity than other `late` hints that style footnotes. (I only learned today that tweak order depends on the name of the tweak, so `priority` is actually necessary).